src/cabal-details.js

```diff
@@ -408,6 +412,66 @@ class CabalDetails extends EventEmitter {
     }
   }
 
+  /**
+   * Archive a channel. Publishes a message announcing
+   * that you have archived the channel, applying it to the views of others who have you as a moderator/admin.
+   * @param {string} channel
+   * @param {string} [reason]
+   * @param {function} cb - callback invoked when the operation has finished, with error as its only parameter
+   */
+  archive (channel, reason = "", cb) {
+    if (!cb) cb = noop
+    const details = this.channels[channel]
+
+    if (channel === '!status') {
+      return nextTick(cb, new Error('cannot archive the !status channel'))
```

Comment:
these errors are getting thrown away in the case where no callback isn't supplied by the user. maybe throw a log message in there to surface the error in the case that the callback is missing?

Reply:
📦